### ISSUE TITLE
MSE Loss - Sigma VAE Pytorch

### ISSUE THREAD

**Issue by adityabantwal** [orybkin/sigma-vae-pytorch#6]
Hi, 
The MSE loss is going towards - infinity so what is the use of the constant term "+ 0.5 * np.log(2 * np.pi)" used in the gaussian_null function. https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L149

**Comment by orybkin** [comment]
Hi Aditya,

Regarding your first question, you need to provide more details. The MSE
loss cannot be going towards -infinity if your variance is bounded. Maybe
double check that you didn't remove this line:
https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L125.
I have never observed this problem using this code.

Regarding your second question, the constant term is part of the
log-likelihood definition for the Gaussian distribution (
https://en.wikipedia.org/wiki/Normal_distribution). You don't have to use
it to train the model since it's a constant.

Oleh

**Comment by orybkin** [comment]
On a separate note, did you manage to get colab to work?

**Comment by adityabantwal** [comment]
Hi Orybkin, 
The colab is working fine with pytorch 1.4, and thanks a lot for your pointers.

**Comment by ndrwmlnk** [comment]
@orybkin 
This constant term  [+ 0.5 * np.log(2 * np.pi)](https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L149) goes straight to the [total_loss](https://github.com/orybkin/sigma-vae-pytorch/blob/b561fc92355e81512a81104cb1e21b06cc9459e2/train_vae.py#L64), which needs to be minimised. Would it make a difference if we just remove this constant term [+ 0.5 * np.log(2 * np.pi)](https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L149)?
E.g.: 
old: `return 0.5 * torch.pow((x - mu) / log_sigma.exp(), 2) + log_sigma + 0.5 * np.log(2 * np.pi)`
new: `return 0.5 * torch.pow((x - mu) / log_sigma.exp(), 2) + log_sigma` 

[total_loss = rec + kl](https://github.com/orybkin/sigma-vae-pytorch/blob/b561fc92355e81512a81104cb1e21b06cc9459e2/train_vae.py#L64)
[rec, kl = model.loss_function(recon_batch, data, mu, logvar)](https://github.com/orybkin/sigma-vae-pytorch/blob/b561fc92355e81512a81104cb1e21b06cc9459e2/train_vae.py#L62)
[rec = self.reconstruction_loss(recon_x, x)](https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L137)
[rec = gaussian_nll(x_hat, log_sigma, x).sum()](https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L127)
[return 0.5 * torch.pow((x - mu) / log_sigma.exp(), 2) + log_sigma + 0.5 * np.log(2 * np.pi)](https://github.com/orybkin/sigma-vae-pytorch/blob/4748a3ac1686316292607f40192c7ec2ded09893/model.py#L149)

**Comment by orybkin** [comment]
Andrew, as I said in my previous message, you don't need to use the
constant term since it doesn't change the gradient.
